### DISCUSSION TITLE
Create/Assign Variable Not Working

### DISCUSSION THREAD

**Discussion by bmillspaugh** [postmanlabs/postman-flows#149] [Defect]
I have a simple flow which calls an API to create a shipment and it returns back a `shipmentId` number in the JSON body that I then use `Assign Variable` to assign it to a `shipment-id` variable, but when I inspect the variable it shows as `<tag:yaml.org,2002:js/undefined>`:

![12-3-2021 5-08-06 PM](https://user-images.githubusercontent.com/3683816/144679643-9e18e284-5fc0-4dee-944f-9a5253c05961.jpg)

In the terminal you can see that the response does include the `shipmentId` number and that the `shipment-id` variable is undefined:

![12-3-2021 5-08-24 PM](https://user-images.githubusercontent.com/3683816/144679844-f5f2b15d-8b60-4c43-8e19-1b75f3a04033.jpg)

I also tried to use the `Create Variable` option, but even though it shows the `shipmentId` in the input I am unable to select it on the `Configuration` page:

![12-3-2021 5-03-48 PM](https://user-images.githubusercontent.com/3683816/144680461-26de18ad-5e00-47b9-bf13-6f368bcce859.jpg)

![12-3-2021 5-03-36 PM](https://user-images.githubusercontent.com/3683816/144680500-c9952ecd-13fd-4300-9915-58fcbc4361b1.jpg)

**Comment by dev-kanishk** [comment]
Hey, after observing  from your image, the response content type header  is `text/html` but body seems to be a JSON string. Send Request block only parses response if the header content type is correctly set to `application/json`. So if you have control over the API, can you try sending the correct content type header?

**Reply by bmillspaugh** [reply]
Thanks @dev-kanishk, that was it!  I thought that it would work because **Postman Flows** is able to parse the JSON content in the `Assign Variables` dialog and it allowed me to select the node, but I guess the flow doesn't use the same logic that the dialog selection uses.

This is not my API so I couldn't fix it myself, but I was able to contact the developer and they already released a new version that returns the correct `Content-Type: application/json` and I confirmed that the flow now works correctly.

Thanks again for your help!
